src/main/java/com/google/sps/servlets/MentorApprovalServlet.java

```diff
@@ -0,0 +1,130 @@
+
+  /**
+   * Returns information about a mentor and the evidence of their internship.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Get IDs of mentor and approver.
+    int mentorId = Utility.tryParseInt(request.getParameter("id"));
+    int approverId = Utility.getUserId();
+    
+    // Set default variables to create MentorEvidence object.
+    boolean isApprover = false;
+    String mentorUsername = "";
+    String paragraph = "";
+
+    if (userService.isUserLoggedIn()) {
+      // If user is logged in, update variables.
+      isApprover = checkForApprover(mentorId, approverId);
+      mentorUsername = Utility.getUsername(mentorId);
+      paragraph = getMentorEvidence(mentorId);
+    }
+
+    MentorEvidence mentorEvidence =
+        new MentorEvidence(isApprover, mentorUsername, paragraph);
+    response.setContentType("application/json");
+    response.getWriter().println(Utility.convertToJsonUsingGson(mentorEvidence));
+  }
+
+  /**
+   * Returns true if approver is assigned to mentee.
+   */
+  private boolean checkForApprover(int mentorId, int approverId) {
+    // Create the MySQL prepared statement, execute it, and store the result.
+    String query = "SELECT * FROM MentorApproval "
+        + "WHERE mentor_id = ? AND approver_id = ?";
+
+    try {
+      // Establish connection to MySQL database.
+      Connection connection = DriverManager.getConnection(
+            Utility.SQL_LOCAL_URL, Utility.SQL_LOCAL_USER, Utility.SQL_LOCAL_PASSWORD);
+      
+      // Create the MySQL SELECT prepared statement.
+      PreparedStatement preparedStatement = connection.prepareStatement(query);
+      preparedStatement.setInt(SqlConstants.MENTOR_APPROVAL_FETCH_MENTORID, mentorId);
+      preparedStatement.setInt(SqlConstants.MENTOR_APPROVAL_FETCH_APPROVERID, approverId);
+      ResultSet queryResult = preparedStatement.executeQuery();
+
+      // If link is found between mentor and approver in MentorApproval table, return true.
+      if (queryResult.next()) {
+        return true;
+      }
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(MentorApprovalServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+    // If no link was found between mentor and approver in MentorApproval table, return false.
+    return false;
+  }
+
+  /**
+   * Returns internship evidence of a mentor.
+   */
+  // TODO(oumontiel): Add more evidence fields.
+  private String getMentorEvidence(int mentorId) {
+    String paragraph = "";
+
+    // Create the MySQL prepared statement, execute it, and store the result.
+    String query = "SELECT * FROM MentorEvidence "
+        + "WHERE mentor_id = ?";
```

Comment:
This can be instantiated within the 'try' block, since it's not used outside that block.

Reply:
Done! I think I do this for every query I use, should I modify it everywhere?

Reply:
Eventually :) As you see them while you're working on future PRs, definitely feel free to update them! It's not hugely important, just a small code / readability optimiziation.